### PR TITLE
Introduce new Gradle build.

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

File: dist/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Weld
 What is it?
 ------------
 
-Weld is the reference implementation of CDI: Contexts and Dependency Injection 
+{CUSTOM_TEXT}: Contexts and Dependency Injection
```
Comment (tremes) [inline dist/README.md]: This is little hack. You can specify your custom text here during script execution (it is replaced with 'sed' command). If you leave the value empty, then the default text ("Weld is the reference implementation of CDI") with cdi version would be included.

File: dist/build.gradle [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+dependencies {
+    compile dep("org.jboss.weld:weld-core", weldVersion)
+    compile dep("org.jboss.weld:weld-core-impl", weldVersion)
+    compile dep("org.jboss.weld.environment:weld-environment-common", weldVersion)
+    compile dep("org.jboss.weld.se:weld-se-core", weldVersion)
+    if(weldVersion.startsWith("3")){
+        compile dep("org.jboss.weld.se:weld-se-shaded", weldVersion)
+        compile dep("org.jboss.weld.servlet:weld-servlet-shaded", weldVersion)
+    }else {
```
Comment (jharting) [inline dist/build.gradle]: formatting

File: dist/build.gradle [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    }
+    compile dep("org.jboss.weld.servlet:weld-servlet-core", weldVersion)
+    // no need for version since compile configuration is transitive by default
+    // listed only because we need to download javadoc and sources artifacts too
+    compile dep("javax.enterprise:cdi-api")
+    compile dep("org.jboss.weld:weld-api")
+    compile dep("org.jboss.weld:weld-spi")
+}
+
+// helper function which return version of transitive dependency according the given artifactId
```
Comment (jharting) [inline dist/build.gradle]: which return**s**

File: dist/build.gradle [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    if(!project.hasProperty("weldVersion"))
+        releaseVersion = new String(console.readLine("\n Weld Core Version (used to locate built Weld jars): \n"))
+    else
+        releaseVersion = weldVersion
+
+    if (!weldDir.exists())
+        throw new FileNotFoundException("Cannot find " + weldDir.absolutePath);
+
+    if (releaseVersion.equals(""))
+        throw new RuntimeException("You must specify valid Weld version.")
```
Comment (jharting) [inline dist/build.gradle]: IllegalArgumentException may be more suitable

**Comment by jharting** [comment]
Merged with minor changes, thanks!

**Pull request closed without merging**